src/main/java/com/jcs/sbs/model/DescribeSnapshotsRequest.java

```diff
@@ -30,57 +39,51 @@ public DescribeSnapshotsRequest(List<String> snapshotsIds) {
     }
 
     /**
-     * This method returns the IDs of the snapshots that is to be described.
+     * Returns the IDs of the snapshots that is to be described.
      * 
      * @return List of snapshotIds
      */
     public List<String> getSnapshotIds() {
-        if (snapshotIds == null) {
-            snapshotIds = new ArrayList<String>();
-        }
         return snapshotIds;
     }
 
     /**
-     * It sets the list of snapshot IDs that are to be described. To describe
-     * all the snapshots, there is no need to set snapshotIds
+     * Sets the list of snapshot IDs that are to be described. To describe all
+     * the snapshots, there is no need to set snapshotIds
      * 
      * @param snapshotIds
      *            The IDs of the snapshot that are to be described.
      */
     public void setSnapshotIds(Collection<String> snapshotIds) {
         if (snapshotIds == null) {
-            this.snapshotIds = null;
+            this.snapshotIds = new ArrayList<String>();
```

Comment:
why do you need this. You initialized snapshots in the constructor itself. I can imagine a case, where someone calls up this function twice. In that scenario the change makes sense. But then you should handle it in withSnapshotIds as well.